vnext/Microsoft.ReactNative/Modules/DeviceInfoModule.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "DeviceInfoModule.h"
+#include <IReactDispatcher.h>
+#include <winrt/Windows.UI.Core.h>
+#include <winrt/Windows.UI.ViewManagement.h>
+
+namespace Microsoft::ReactNative {
+
+DeviceInfoHolder::DeviceInfoHolder() {
+  updateDeviceInfo();
+}
+
+void DeviceInfoHolder::InitDeviceInfoHolder(
+    const winrt::Microsoft::ReactNative::ReactPropertyBag &propertyBag) noexcept {
+  auto deviceInfoHolder = std::make_shared<DeviceInfoHolder>();
+  deviceInfoHolder->updateDeviceInfo();
+
+  propertyBag.Set(
+      winrt::Microsoft::ReactNative::ReactPropertyId<
+          winrt::Microsoft::ReactNative::ReactNonAbiValue<std::shared_ptr<DeviceInfoHolder>>>(
+          L"DeviceInfo", L"DeviceInfoHolder"),
+      std::move(deviceInfoHolder));
+
+  auto const &displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
+  auto const &window = xaml::Window::Current().CoreWindow();
```

Comment:
> xaml::Window::Current().CoreWindow(); [](start = 22, length = 38)

I wonder if it causes issues with the XAML islands.

Reply:
I'll check.  But it that same as what we were doing before.   I think there is a CoreWindow shim used for islands.

Reply:
Then it should be fine, I guess.

---
In reply to: [429371981](https://github.com/microsoft/react-native-windows/pull/4991#discussion_r429371981) [](ancestors = 429371981)